src/plugins/intel_gpu/src/runtime/kernels_cache.cpp

```diff
@@ -327,12 +344,22 @@ void kernels_cache::build_batch(const engine& build_engine, const batch_program&
 
             program.createKernels(&kernels);
 
+            auto program_bin = getProgramBinaries(program);
+
+            // save to in_memory cache
+            // Cache is saved at both in-memory and file, because in-memory cache capacity is limited.
+            if (batch.kernels_counter == 1) {
+                _in_memory_kernels_cache.get(batch.hash_value, [program_bin](){
+                    return LRUCache<size_t, std::vector<uint8_t>>::CacheEntry{program_bin, program_bin.size()};
```

Comment:
I think, if in_memory cache is turned on, we'd better to save to permanent cache only when the cache entry is created. even though it is write-through policy. 

So..
if (in_memory_cache) { 
     in_memory_cache.get(key, [&]() {
         if (is_cache_enabled()) { // both in_mem cache & permanant cache used
             saveBinarytoFile();
         }
         return LRUCache<...>()};
     } else if (is_cache_enabled) { // only permanent cache used 
         saveBinaryToFile(..);
    }
}